### ISSUE TITLE
TS project wide doesn't work on safari

### ISSUE THREAD

**Issue by jrieken** [microsoft/vscode#175229]
* use safari
* open https://insiders.vscode.dev/github/microsoft/vscode-markdown-languageservice?vscode-coi=
* open index.ts
* 🐛 TS initialises forever, console shows the error below

```
[Error] Unhandled Promise Rejection: TypeError: undefined is not an object (evaluating 'f')
	(anonymous function) (tsserver.web.js:1:3637082)
	asyncFunctionResume
	dispatchEvent
	(anonymous function) (11a837e0-9d6d-4655-927a-465d3a6a4e5c:1:270)
```

**Comment by mjbvz** [comment]
After #176294, this now woks fine in Safari tech preview. I suspect there's an issue with our nested worker polyfil but am still looking into this

**Comment by mjbvz** [comment]
I've tried debugging what is going on here but can't figure it out (and safari's dev tools are certainly not helping)

What I suspect is that our nested worker polyfill causes a message to be dropped somewhere, possibly because a message port is started early or never started

Given that everything works in safari tech preview though and given that don't plan on shipping project wide intellisense until cross origin isolation is enabled, this is a lower priority. Un-assigning from this month

**Comment by mjbvz** [comment]
I believe this all works in Safari 16.4 which added nested worker support. Not sure what the issue was with our nested worker polyfill (note that #183482 needs to be merged too to see this working end to end)
